src/Page.js

```diff
@@ -830,7 +806,8 @@ Page.prototype.preRender = function (content) {
   Object.entries(this.plugins).forEach(([pluginName, plugin]) => {
     if (plugin.preRender) {
       preRenderedContent
-        = plugin.preRender(preRenderedContent, this.pluginsContext[pluginName] || {}, this.frontMatter);
+        = plugin.preRender(preRenderedContent, this.pluginsContext[pluginName] || {},
+                           this.frontMatter, this.pageConfig);
     }
```

Comment:
Passing `this.pageConfig` will allow plugins to modify the pageConfig object owned by the `Page`. While i understand `this.pageConfig` is only used in plugins, I'm thinking it would be better to construct a copy in the pre and post render functions explicitly.

Perhaps before the foreach:

```js
const config = {
  rootPath: this.rootPath,
  enableSearch: this.enableSearch,
};
```

This allows us to explicitly control what plugins have access to and prevents them from changing the object owned by `Page`. However, it does not prevent plugins from changing the config being passed to other plugins.

Reply:
And then this line can be removed:
https://github.com/MarkBind/markbind/blob/7c8dfbb945b2f6c3f4a82c3caac7df6c4a6f3c9f/src/Page.js#L68

Reply:
What if we made a copy before passing it in?